### ISSUE TITLE
Specification for ORAS attach command

### ISSUE THREAD

**Issue by shizhMSFT** [oras-project/oras#424]
### Discussed in https://github.com/oras-project/oras/discussions/407

<div type='discussions-op-text'>

<sup>Originally posted by **sajayantony** June 10, 2022</sup>
Currently the ORAS CLI has a command that pushes an reference type using the subject flag 

```bash
oras push localhost:5000/alpine \
   --subject localhost:5000/alpine:3.9 \
   --artifact-type doc/example --plain-http \
    ./README.md
```

I would like to recommend moving this under a new command group `attach` and credit goes to @jdolitsky for showing this in the OCI working group. 

The main idea idea here would be to support both the specification oras-artifacts and OCI reftypes. 
The behavior can be determined using a few options - 

1. Querying the extension API to see if we need to support ORAS or use OCI reftype 
2. User passes in a flag like --update-index as per the propsal that might get finally approved. 

For e.g. 

```bash
oras attach --artifact-type doc/example \
       ./README.md \
        localhost:5000/alpine:3.9 \
```
 
@FeynmanZhou @shizhMSFT </div>

**Comment by qweeah** [comment]
Ack

**Comment by shizhMSFT** [comment]
@qweeah It is related to #326 and #379

**Comment by qweeah** [comment]
Need to support annotation for `oras attach`

**Comment by shizhMSFT** [comment]
Closed by #433
